presto-main/src/test/java/com/facebook/presto/sql/TestExpressionInterpreter.java

```diff
@@ -779,6 +779,22 @@ public void testCastOptimization()
         assertOptimizedMatches("cast(unbound_string as VARCHAR)", "unbound_string");
         assertOptimizedMatches("cast(unbound_integer as INTEGER)", "unbound_integer");
         assertOptimizedMatches("cast(unbound_string as VARCHAR(10))", "cast(unbound_string as VARCHAR(10))");
+
+        // nested varchar
+        assertRowExpressionEquals(OPTIMIZED, "cast( cast(unbound_string as VARCHAR) as VARCHAR)", "unbound_string");
```

Comment:
Can you also add tests to show that for other optimization levels the input is unchanged? Thanks!

Reply:
Thanks for the quick review!
I added tests for `SERIALIZABLE` level.